### PR TITLE
Add support for shiny mode magic comment

### PR DESCRIPTION
**Pull request by wch**
## Intent

This PR adds support for shiny mode magic comments which are generated by interactive Quarto dashboards that use Shiny for Python. This change allows users to create Quarto dashboards that have `from shiny.express import ...` instead of `from shiny ...`; previously those apps would not run.

See the following for more information:
- https://github.com/posit-dev/py-shiny/pull/1217 
- https://github.com/quarto-dev/quarto-web/pull/1049

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach

This copies over code from the py-shiny repository, from this file:

https://github.com/posit-dev/py-shiny/blob/2b66c898cb9d0fc827e587b11a839e270b2e814f/shiny/express/_is_express.py

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

**Comment by wch** [comment]
The linting test is failing because in Python 3.7, the `typing` module doesn't have `Literal`. (It is present in 3.8+.)

I could work around this by importing from `typing_extensions`. That said, my understanding is that we've dropped Python 3.7 support, so can we stop testing on that platform?

**Comment by mmarchetti** [comment]
rsconnect-python hasn't removed [Python 3.7 support](https://github.com/rstudio/rsconnect-python/blob/master/pyproject.toml#L8), but Connect has removed it so it's probably time.

**Comment by nealrichardson** [comment]
> rsconnect-python hasn't removed [Python 3.7 support](https://github.com/rstudio/rsconnect-python/blob/master/pyproject.toml#L8), but Connect has removed it so it's probably time.

I made https://github.com/rstudio/rsconnect-python/issues/556 for that.

File: rsconnect/shiny_express.py [hunk 1]
```diff
@@ -56,25 +66,52 @@ def __init__(self):
...
+    def visit_Module(self, node: ast.Module) -> None:
         super().generic_visit(node)
 
     # Don't recurse into any nodes, so the we'll only ever look at top-level nodes.
-    def generic_visit(self, node: ast.AST):
+    def generic_visit(self, node: ast.AST) -> None:
         pass
 
 
+def find_magic_comment_mode(content: str) -> Literal["core", "express"] | None:
```
Comment (nealrichardson) [inline rsconnect/shiny_express.py:88]: Would it be worth adding a test for this function?

**Comment by aronatkins** [comment]
@wch - if you pull in master, you should see CI pass.

File: rsconnect/shiny_express.py (hunk 1 above)
Comment (wch) [reply rsconnect/shiny_express.py:88]: I've added tests for Shiny Express app detection in general in [cc3766e](https://github.com/rstudio/rsconnect-python/pull/549/commits/cc3766e3b30b5863a309153020e736c251cff6aa).

**Pull request merged**